src/pyhf/contrib/utils.py

```diff
@@ -0,0 +1,67 @@
+"""Helper utilities for common tasks."""
+
+from urllib.parse import urlparse
+import tarfile
+from io import BytesIO
+import logging
+from .. import exceptions
+
+logging.basicConfig()
+log = logging.getLogger(__name__)
+
+try:
+    import requests
+
+    def download(archive_url, output_directory, force=False, compress=False):
+        """
+        Download the patchset archive from the remote URL and extract it in a
+        directory at the path given.
+
+        Example:
+
+            >>> from pyhf.contrib.utils import download
+            >>> download("https://www.hepdata.net/record/resource/1408476?view=true", "1Lbb-likelihoods")
+            >>> import os
+            >>> os.listdir("1Lbb-likelihoods")
+            ['patchset.json', 'README.md', 'BkgOnly.json']
```

Comment:
doctest is failing as the order is random. So I need to let doctest know that it should expect a list like this, but not care about the order. Not sure if this is easy/possible.